cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -396,67 +448,75 @@ def create_rna_dna_sample_case_map(self, rna_case: models.Family) -> Dict[str, D
         Case Returns:
             rna_dna_sample_case_map     (Dict):       rna-dna relationships, and related dna cases based on subject id
         """
-        rna_dna_sample_case_map: Dict[str, Dict[str, list]] = {}
-        for rna_case.link in rna_case.links:
-            self._add_rna_sample(
-                rna_sample=rna_case.link.sample, rna_dna_sample_case_map=rna_dna_sample_case_map
+        rna_dna_sample_case_map: Dict[str, Dict[str, List[str]]] = {}
+        for link in rna_case.links:
+            self._map_rna_sample(
+                rna_sample=link.sample, rna_dna_sample_case_map=rna_dna_sample_case_map
             )
         return rna_dna_sample_case_map
 
-    def _add_rna_sample(
-        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, list]]
-    ) -> Dict[str, Dict[str, list]]:
-        """Adds an RNA sample and its matching DNA sample, and cases."""
-        dna_sample: models.Sample = self._link_rna_sample_to_dna_sample(
+    def _map_rna_sample(
+        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, List[str]]]
+    ) -> Dict[str, Dict[str, List[str]]]:
+        """Create a dictionary of all DNA samples, and their related cases, related to a RNA sample."""
+        dna_sample: models.Sample = self._map_dna_samples_related_to_rna_sample(
             rna_sample=rna_sample, rna_dna_sample_case_map=rna_dna_sample_case_map
         )
-        self._add_dna_cases_to_dna_sample(
+        self._map_dna_cases_to_dna_sample(
             dna_sample=dna_sample,
             rna_dna_sample_case_map=rna_dna_sample_case_map,
             rna_sample=rna_sample,
         )
         return rna_dna_sample_case_map
 
-    def _link_rna_sample_to_dna_sample(
-        self, rna_sample: models.Sample, rna_dna_sample_case_map: Dict[str, Dict[str, list]]
+    def _map_dna_samples_related_to_rna_sample(
```

Comment:
This one is complex. Would be nice with some unit tests to make sure it does what it supposed to.

Reply:
done